tests/consent/api/test_views.py

```diff
@@ -0,0 +1,452 @@
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+                'exists': True,
+                'consent_provided': True,
+                'consent_required': False
+            },
+            200
+        ),
+        # Consent not given for an enterprise course enrollment.
+        (
+            factories.EnterpriseCourseEnrollmentFactory,
+            [{
+                'enterprise_customer_user__user_id': TEST_USER_ID,
+                'enterprise_customer_user__enterprise_customer__uuid': TEST_UUID,
+                'consent_granted': False,
+                'course_id': TEST_COURSE,
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+                'exists': True,
+                'consent_provided': False,
+                'consent_required': True
+            },
+            200
+        ),
+        # Consent given for an audit.
+        (
+            factories.UserDataSharingConsentAuditFactory,
+            [{
+                'user__user_id': TEST_USER_ID,
+                'user__enterprise_customer__uuid': TEST_UUID,
+                'state': 'enabled'
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+                'exists': True,
+                'consent_provided': True,
+                'consent_required': True
+            },
+            200
+        ),
+        # Consent not given for an audit.
+        (
+            factories.UserDataSharingConsentAuditFactory,
+            [{
+                'user__user_id': TEST_USER_ID,
+                'user__enterprise_customer__uuid': TEST_UUID,
+                'state': 'disabled'
+            }],
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE
+            },
+            {
+                'username': TEST_USERNAME,
+                'enterprise_customer': TEST_UUID,
+                'course_id': TEST_COURSE,
+                'exists': True,
+                'consent_provided': False,
+                'consent_required': True
+            },
+            200
+        ),
+    )
+    @ddt.unpack
+    def test_consent_api_get_endpoint(self, factory, items, request_body, expected_response_body, expected_status_code):
```

Comment:
Let's add checks for the different cases of EnterpriseCustomer consent requirement as well.